daphne/src/taskprov.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{
+    messages::{
+        taskprov::{VdafType, TASK_PROV_SALT},
+        Id,
+    },
+    vdaf::VdafVerifyKey,
+};
+use ring::{
+    error::Unspecified,
+    hkdf::{Prk, Salt, HKDF_SHA256},
+};
+
+pub fn compute_vdaf_verify_prk(verify_key_init: &[u8]) -> Prk {
+    // The documentation says computing the Salt is expensive, and we use the same PRK all the
+    // time, so we compute it once.
+    Salt::new(HKDF_SHA256, &TASK_PROV_SALT).extract(verify_key_init)
+}
+
+/// Compute the VDAF verify key for task_id and the specified VDAF type.
+pub fn compute_vdaf_verify_key(
+    prk: &Prk,
+    task_id: &Id,
+    vdaf_type: VdafType,
+) -> Result<VdafVerifyKey, Unspecified> {
+    let info = [task_id.as_ref()];
+    let okm = prk.expand(&info, vdaf_type.clone())?;
+    match &vdaf_type {
+        VdafType::Prio3Aes128Count | VdafType::Prio3Aes128Sum | VdafType::Prio3Aes128Histogram => {
+            let mut bytes = [0u8; 16];
+            okm.fill(&mut bytes[..])?;
```

Comment:
Unless the error here depends on the input, I don't think propagating this error is all that useful. Eventually we'd need to handle this as a `DapAbort`.
```
            okm.fill(&mut bytes[..]).expect("HKDF expand failed");
```